include/lua.d.ts

```diff
@@ -101,7 +101,7 @@ interface LuaMetatable<T> {
 }
 
 /** Sets the metatable for the given table. If `metatable` is nil, the metatable of the given table is removed. If the original metatable has a "__metatable" field, this will raise an error. This function returns the table t, which was passed to the function. */
-declare function setmetatable<T extends object>(object: T, metatable: LuaMetatable<T>): T;
+declare function setmetatable<T extends object>(object: T, metatable: LuaMetatable<T> | undefined): T;
```

Comment:
Should this just be an optional argument instead?

```suggestion
declare function setmetatable<T extends object>(object: T, metatable?: LuaMetatable<T>): T;
```

Reply:
![image](https://github.com/user-attachments/assets/7497c18f-176e-4e65-8e4e-31ba79393604)